source/lmp/pair_nnp.cpp

```diff
@@ -105,6 +105,54 @@ int PairNNP::get_node_rank() {
     return looprank;
 }
 
+std::string PairNNP::get_file_content(const std::string & model) {
+  int myrank = 0, root = 0;
+  MPI_Comm_rank(MPI_COMM_WORLD, &myrank);
+  unsigned nchar = 0;
+  std::string file_content;
+  if (myrank == root) {
+    checkStatus (ReadFileToString(Env::Default(), model, &file_content));
+    nchar = file_content.size();
+  }
+  MPI_Bcast(&nchar, 1, MPI_UNSIGNED, root, MPI_COMM_WORLD);  
+  char * buff = (char *)malloc(sizeof(char) * nchar);  
+  if (myrank == root) {  
+    memcpy(buff, file_content.c_str(), sizeof(char) * nchar);
+  }
+  MPI_Bcast(buff, nchar, MPI_CHAR, root, MPI_COMM_WORLD);
+  file_content.resize(nchar);
+  for (unsigned ii = 0; ii < nchar; ++ii) {
+    file_content[ii] = buff[ii];
+  }
+  return file_content;
+  free(buff);
+}
+
+std::vector<std::string> PairNNP::get_file_content(const std::vector<std::string> & models) {
+  int myrank = 0, root = 0;
+  MPI_Comm_rank(MPI_COMM_WORLD, &myrank);
+  unsigned nchar = 0;
+  std::vector<std::string> file_contents(models.size());
+  for (unsigned ii = 0; ii < models.size(); ++ii) {
+    if (myrank == root) {
+      checkStatus (ReadFileToString(Env::Default(), models[ii], &file_contents[ii]));
+      nchar = file_contents[ii].size();
+    }
+    MPI_Bcast(&nchar, 1, MPI_UNSIGNED, root, MPI_COMM_WORLD);  
+    char * buff = (char *)malloc(sizeof(char) * nchar);  
+    if (myrank == root) {  
+      memcpy(buff, file_contents[ii].c_str(), sizeof(char) * nchar);
+    }
+    MPI_Bcast(buff, nchar, MPI_CHAR, root, MPI_COMM_WORLD);
+    file_contents[ii].resize(nchar);
+    for (unsigned jj = 0; jj < nchar; ++jj) {
+      file_contents[ii][jj] = buff[jj];
+    }
+    free(buff);
```

Comment:
This section of code should reuse the same piece in `std::string PairNNP::get_file_content(const std::string & model)`